internal/sbom/spdx.go

```diff
@@ -40,82 +38,189 @@ func GenerateSPDX(koVersion string, date time.Time, mod []byte, imgDigest v1.Has
+	})
+
+	doc.Packages = append(doc.Packages, Package{
+		Name: bi.Main.Path,
+		ID:   mainPackageID,
+		// TODO: PackageSupplier: "Organization: " + bs.Main.Path
+		DownloadLocation: "https://" + bi.Main.Path,
+		FilesAnalyzed:    false,
+		// TODO: PackageHomePage:  "https://" + bi.Main.Path,
+		LicenseConcluded: "NOASSERTION",
+		LicenseDeclared:  "NOASSERTION",
+		// TODO: PackageLicenseComments: "NOASSERTION"
+		CopyrightText: "NOASSERTION",
+		// TODO: PackageComment: "NOASSERTION"
+		ExternalRefs: []ExternalRef{{
+			Category: "PACKAGE_MANAGER",
+			Type:     "purl",
+			Locator:  ociRef(bi.Path, imgDigest),
+		}},
+	})
+
+	for _, dep := range bi.Deps {
+		depID := fmt.Sprintf("SPDXRef-Package-%s-%s",
+			strings.ReplaceAll(dep.Path, "/", "."),
+			dep.Version)
+
+		doc.Relationships = append(doc.Relationships, Relationship{
+			Element: mainPackageID,
+			Type:    "DEPENDS_ON",
+			Related: depID,
+		})
+
+		pkg := Package{
+			Name:    dep.Path,
+			ID:      depID,
+			Version: dep.Version,
+			// TODO: PackageSupplier: "Organization: " + dep.Path
+			DownloadLocation: fmt.Sprintf("https://proxy.golang.org/%s/@v/%s.zip", dep.Path, dep.Version),
+			FilesAnalyzed:    false,
+			LicenseConcluded: "NOASSERTION",
+			LicenseDeclared:  "NOASSERTION",
+			// TODO: PackageLicenseComments: "NOASSERTION"
+			CopyrightText: "NOASSERTION",
+			// TODO: PackageComment: "NOASSERTION"
+			ExternalRefs: []ExternalRef{{
+				Category: "PACKAGE_MANAGER",
+				Type:     "purl",
+				Locator:  goRef(dep.Path, dep.Version),
+			}},
+		}
+
+		if dep.Sum != "" {
+			pkg.Checksums = []Checksum{{
+				Algorithm: "SHA256",
+				Value:     h1ToSHA256(dep.Sum),
+			}}
+		}
+
+		doc.Packages = append(doc.Packages, pkg)
+	}
+
 	var buf bytes.Buffer
-	if err := tmpl.Execute(&buf, tmplInfo{
-		BuildInfo: *bi,
-		Date:      date.Format(dateFormat),
-		KoVersion: koVersion,
-		ImgDigest: imgDigest,
-	}); err != nil {
+	enc := json.NewEncoder(&buf)
+	enc.SetIndent("", "  ")
+	if err := enc.Encode(doc); err != nil {
 		return nil, err
 	}
 	return buf.Bytes(), nil
 }
 
-type tmplInfo struct {
-	BuildInfo
-	Date, UUID, KoVersion string
-	ImgDigest             v1.Hash
+// Below this is forked from here:
+// https://github.com/kubernetes-sigs/bom/blob/main/pkg/spdx/json/v2.2.2/types.go
+
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+const (
+	NOASSERTION = "NOASSERTION"
+	Version     = "SPDX-2.2"
+)
+
+type Document struct {
```

Comment:
Nit since it's internal anyway but these types can be unexported

Reply:
I was trying to avoid touching this code too much.  I was about to change, but remembered this is in `internal/` so I'm sort of inclined to leave it.  WDYT?